thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/email/filter/BaseAlertFilter.java

```diff
@@ -8,6 +8,19 @@
 
 public abstract class BaseAlertFilter implements AlertFilter {
   private final static Logger LOG = LoggerFactory.getLogger(BaseAlertFilter.class);
+
+  // Check if the given string can be parsed as double value
+  public static boolean isNumeric(String str) {
```

Comment:
Probably "NumberUtils.isParsable" is a good choice too.
https://stackoverflow.com/questions/1102891/how-to-check-if-a-string-is-numeric-in-java